scheduler/cloudformation/cloudformation.go

```diff
@@ -268,11 +271,129 @@ func (s *Scheduler) submit(ctx context.Context, tx *sql.Tx, app *scheduler.App,
 		if err := <-done; err != nil {
 			return err
 		}
+		done = make(chan error, 1)
+		s.waitForDeploymentToStabilize(ctx, stackName, ss, done)
+		if err = <-done; err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
+func (s *Scheduler) waitForDeploymentToStabilize(ctx context.Context, name string, ss scheduler.StatusStream, done chan error) {
```

Comment:
Also, minor, maybe `waitForDeploymentsToStabilize` (plural `Deployments`) since we're waiting on multiple deployments.